sources/nvidia-migmanager/src/main.rs

```diff
@@ -0,0 +1,697 @@
+        || pci_device_id.starts_with("0x2321")
+        || pci_device_id.starts_with("0x2331")
+        || pci_device_id.starts_with("0x2339")
+    {
+        info!("Found NVIDIA H100-80GB GPU.");
+        Ok(NvidiaGpu::H100_80GB)
+    } else if pci_device_id.starts_with("0x2335")
+        || pci_device_id.starts_with("0x233B")
+        || pci_device_id.starts_with("0x2348")
+    {
+        info!("Found NVIDIA H200-141GB GPU.");
+        Ok(NvidiaGpu::H200_141GB)
+    } else {
+        warn!("Found NVIDIA Device but couldn't confirm variant.");
+        Ok(NvidiaGpu::Other)
+    }
+}
+
+fn get_gpu_state(current_state: &str, next_state: &str) -> MigState {
+    let mut gpu_state = MigState::Unknown;
+
+    if current_state != next_state {
+        gpu_state = MigState::Transition;
+    } else if current_state == "Enabled" {
+        gpu_state = MigState::Enabled;
+    } else if current_state == "Disabled" {
+        gpu_state = MigState::Disabled;
+    } else if current_state == "[N/A]" {
+        gpu_state = MigState::Unsupported;
+    }
+
+    gpu_state
+}
+
+// Runs nvidia-smi command to find out the current state of the Nvidia GPU.
+fn get_gpu_info() -> Result<Vec<MigGpu>> {
+    info!("Fetching GPU devices data ...");
+
+    let output = command(
+        NVIDIA_SMI_PATH,
+        [
+            "--query-gpu=pci.device_id,mig.mode.current,mig.mode.pending",
+            "--format=csv,noheader",
+        ],
+    )?;
+
+    let mut gpu_info = Vec::new();
+
+    for row in output.lines() {
+        let parts: Vec<_> = row.split(", ").collect();
+
+        ensure!(parts.len() == 3, error::NvidiaSmiSnafu);
+
+        let gpu_model = get_gpu_model(parts[0])?;
+        let gpu_state = get_gpu_state(parts[1], parts[2]);
+
+        let gpu = MigGpu {
+            model: gpu_model,
+            state: gpu_state,
+        };
+        gpu_info.push(gpu);
+    }
+
+    Ok(gpu_info)
+}
+
+/// Read the config file to get MIG settings
+fn get_mig_settings<P>(config_path: P) -> Result<NvidiaMigConfig>
+where
+    P: AsRef<Path>,
+{
+    let config_str = fs::read_to_string(config_path.as_ref()).context(error::ReadConfigSnafu {
+        config_path: config_path.as_ref(),
+    })?;
+
+    let config: NvidiaMigConfig =
+        toml::from_str(&config_str).context(error::TomlDeserializationSnafu {
+            config_path: config_path.as_ref(),
+        })?;
+
+    Ok(config)
+}
+
+fn process_mig_config<T>(model_str: &str, mig_settings: &NvidiaMigConfig) -> Result<()>
+where
+    T: MigGpuProfile,
+{
+    let default_profile = "1".to_string();
+    let gpu_model = model_str.to_string();
+    let mig_profile = mig_settings
+        .profile
+        .get(&gpu_model)
+        .unwrap_or(&default_profile);
+
+    info!("MIG Profile or the number of GPU slices: {:?}", mig_profile);
+    let profile =
+        serde_plain::from_str::<T>(mig_profile.as_str()).context(error::DeserializationSnafu)?;
+
+    set_mig_profile(profile.get_mig_profile())?;
+
+    Ok(())
```

Comment:
nit: can be written as
```suggestion
    set_mig_profile(profile.get_mig_profile())
```